domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -570,6 +571,12 @@ class ExplorationProgressController @Inject constructor(
         recomputeCurrentStateAndNotifyAsync()
       }.launchIn(CoroutineScope(backgroundCoroutineDispatcher))
       explorationProgress.advancePlayStageTo(LOADING_EXPLORATION)
+      explorationProgressListeners.forEach {
```

Comment:
I suggest adding tests to ``ExplorationProgressControllerTest`` to verify both the session start & end cases to make sure that bound listeners are called when expected and passed the correct data.

Reply:
I added one test for this

Reply:
I've reviewed the added test and it covers the required cases. This should be good to go.